[ews-build.webkit.org] Use super for parent class invocations

JonWBedard opened the pull request:
#### 419bb9a7d871315e65087dc79d8d0971db5acb82
<pre>
[ews-build.webkit.org] Use super for parent class invocations
<a href="https://bugs.webkit.org/show_bug.cgi?id=250799">https://bugs.webkit.org/show_bug.cgi?id=250799</a>
rdar://104400449

Reviewed by Aakash Jain.

As we change classes to new-style for our work migrating to Twisted
defer, we should have classes invoke methods via &apos;super&apos; so that
code breaks quickly if we change the parent class.

* Tools/CISupport/ews-build/steps.py:
(ShowIdentifier.__init__): Use super instead of directly invoking a class method.
(ShowIdentifier.start): Ditto.
(ShowIdentifier.evaluateCommand): Ditto.
(ShowIdentifier.getResultSummary): Ditto.
(FindModifiedLayoutTests.__init__): Ditto.
(ValidateChange): Ditto.
(RunJavaScriptCoreTests.__init__): Ditto.
(RunJavaScriptCoreTests.start): Ditto.
(RunJavaScriptCoreTests.evaluateCommand): Ditto.
(RunJavaScriptCoreTests.commandComplete): Ditto.
(RunJavaScriptCoreTests.getResultSummary): Ditto.
(RunJSCTestsWithoutChange.evaluateCommand): Ditto.
(RunWebKitTests.__init__): Ditto.
(RunWebKitTests.start): Ditto.
(RunWebKitTests.runCommand): Ditto.
(UploadBuiltProduct.__init__): Ditto.
(TransferToS3.__init__): Ditto.
(DownloadBuiltProductFromMaster.__init__): Ditto.
(UploadTestResults.__init__): Ditto.
(ExtractTestResults.__init__): Ditto.
(ApplyWatchList.__init__): Ditto.

Canonical link: <a href="https://commits.webkit.org/259233@main">https://commits.webkit.org/259233@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7417b647ca4be7a8983a6ccef5d94573c888a1dd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104400 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13479 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37310 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113616 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/173910 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108323 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14580 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4395 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96627 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/112655 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110167 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/11239 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/94304 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/38864 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93103 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/25915 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/80533 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6839 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/27272 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/6969 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/3821 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/103244 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12995 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/46830 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/8763 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3364 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -2976,15 +2976,16 @@ def getResultSummary(self):
     def evaluateCommand(self, cmd):
-        rc = shell.Test.evaluateCommand(self, cmd)
+        # We're delibrately triggering the version from our parent class's parent class
+        rc = super(RunJavaScriptCoreTests, self).evaluateCommand(cmd)
```
This looks confusing. If we are invoking parent of parent class, someone reading the code wouldn't easily know which class is really being invoked, they would have to move around and open the parent's class code, just to know this, earlier it was clearly visible right here.
Better to skip the change in this class. Same for other instances where we are referring to parent of parent class.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3514,7 +3515,8 @@ def evaluateCommand(self, cmd):
     def runCommand(self, command):
-        yield shell.Test.runCommand(self, command)
+        # We're delibrately triggering the version from our parent class's parent class
+        yield super(RunWebKitTests, self).runCommand(command)
```
Ditto.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3564,14 +3566,16 @@ def start(self):
 
     def evaluateCommand(self, cmd):
-        rc = shell.Test.evaluateCommand(self, cmd)
+        # We're delibrately triggering the version from our parent class's parent class
```
Ditto.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3564,14 +3566,16 @@ def start(self):
     @defer.inlineCallbacks
     def runCommand(self, command):
-        yield shell.Test.runCommand(self, command)
+        # We're delibrately triggering the version from our parent class's parent class
```
Ditto

aj062 reviewed (commented):
Mostly looks good.
Would be nice if we can use Python 3 feature and avoid explicitly listing class name in super().

aj062 reviewed (approved):
r+ with comments

JonWBedard committed:
[ews-build.webkit.org] Use super for parent class invocations
https://bugs.webkit.org/show_bug.cgi?id=250799
rdar://104400449

Reviewed by Aakash Jain.

As we change classes to new-style for our work migrating to Twisted
defer, we should have classes invoke methods via 'super' so that
code breaks quickly if we change the parent class.

* Tools/CISupport/ews-build/steps.py:
(ShowIdentifier.__init__): Use super instead of directly invoking a class method.
(ShowIdentifier.start): Ditto.
(ShowIdentifier.evaluateCommand): Ditto.
(ShowIdentifier.getResultSummary): Ditto.
(FindModifiedLayoutTests.__init__): Ditto.
(ValidateChange): Ditto.
(RunJavaScriptCoreTests.__init__): Ditto.
(RunJavaScriptCoreTests.start): Ditto.
(RunJavaScriptCoreTests.evaluateCommand): Ditto.
(RunJavaScriptCoreTests.commandComplete): Ditto.
(RunJavaScriptCoreTests.getResultSummary): Ditto.
(RunJSCTestsWithoutChange.evaluateCommand): Ditto.
(RunWebKitTests.__init__): Ditto.
(RunWebKitTests.start): Ditto.
(RunWebKitTests.runCommand): Ditto.
(UploadBuiltProduct.__init__): Ditto.
(TransferToS3.__init__): Ditto.
(DownloadBuiltProductFromMaster.__init__): Ditto.
(UploadTestResults.__init__): Ditto.
(ExtractTestResults.__init__): Ditto.
(ApplyWatchList.__init__): Ditto.

Canonical link: https://commits.webkit.org/259233@main
